pandas/core/arrays/_mixins.py

```diff
@@ -168,3 +172,22 @@ def _validate_setitem_key(self, key):
 
     def _validate_setitem_value(self, value):
         return value
+
+    def __getitem__(self, key):
+        if lib.is_integer(key):
+            # fast-path
+            result = self._ndarray[key]
+            if self.ndim == 1:
+                return self._box_func(result)
+            return self._from_backing_data(result)
+
+        key = self._validate_getitem_key(key)
+        result = self._ndarray[key]
+        if lib.is_scalar(result):
+            return self._box_func(result)
+
+        result = self._from_backing_data(result)
+        return result
+
+    def _validate_getitem_key(self, key):
```

Comment:
this duplicates `_validate_setitem_key`. is this needed?

in what cases could one of a getitem and a setitem key be valid and other invalid?

Reply:
DatetimeLikeArrayMixin has special logic here.  #36210 is about deprecating that